### PR TITLE
For documentation, change xelatex invocation from -interaction=nonstopmode to -interaction=batchmode

### PR DESCRIPTION
**Pull request by michael-okeefe**
Work completed under the Big Ladder Software Contribution License Agreement.

Change the documentation building flag in CMake from

```
-interaction=nonstopmode
```

to

```
-interaction=batchmode
```

Both do not halt on warnings or errors but the second flag will not
print out to the command-line. Since LaTeX gives us the build logs for
error reporting, the command line print-out is redundant but I realize
it may be useful in conjunction with the CI.

I did a highly informal test of two cases with the current flag and the
batchmode flag on my mac and build times are slightly faster with the
"batchmode" flag:

```
-interaction=nonstopmode

real	2m56.202s
user	8m37.940s
sys	0m14.456s
```

```
-interaction=batchmode

real	2m32.310s
user	7m42.362s
sys	0m10.172s
```

Potential downsides are that CI may want to process error messages from
stdin. Could a python script be added that checks the LaTeX logs for
errors and reports a succinct list of issues such as errors and
"multiply defined labels"?

Pull request overview
---------------------
Please change this line to a description of the pull request, with useful supporting information including whether it is a new feature, or fixes a defect, a cross reference to any defects addressed in this PR, the type of changes to be made, and whether diffs are expected/justified based on this change.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by nrel-bot-3** [comment]
@michael-okeefe @lgentile it has been 15 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
Do you think we should also add `-halt-on-error` to the TeX command line?  That way it will not always quit very early in addition to never asking the user for interaction.

**Comment by michael-okeefe** [comment]
That sounds like a good idea to add "-halt-on-error". I'd support that.

**Comment by nrel-bot-3** [comment]
@michael-okeefe @lgentile it has been 14 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
So, CI is indeed reading errors from stdout, and using it to report somewhat meaningful errors.  How about we change this PR so that:

- There is CMake logic to switch between nonstopmode and batchmode
- The default is to use batchmode for developers
- The CI yaml settings are adjusted so that it uses nonstopmode
- We add -halt-on-error to the TeX command line

Thoughts?

**Comment by michael-okeefe** [comment]
That works for me and sounds like a good plan. Thanks.

File: .decent_ci-Linux.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ compilers:
...
     release_build_cmake_extra_flags: -DPROFILE_GENERATE:BOOL=OFF -DPROFILE_USE:BOOL=ON -DENABLE_REGRESSION_TESTING:BOOL=ON -DBUILD_DOCS:BOOL=ON
-    cmake_extra_flags: -DBUILD_DOCS:BOOL=ON -DBUILD_FORTRAN=ON -DBUILD_PACKAGE:BOOL=ON -DBUILD_TESTING:BOOL=ON -DENABLE_REGRESSION_TESTING:BOOL=ON -DREGRESSION_BASELINE_PATH:PATH=$REGRESSION_BASELINE -DREGRESSION_SCRIPT_PATH:PATH=$REGRESSION_DIR -DREGRESSION_BASELINE_SHA:STRING=$REGRESSION_BASELINE_SHA -DCOMMIT_SHA=$COMMIT_SHA -DENABLE_GTEST_DEBUG_MODE:BOOL=OFF -DBUILD_PERFORMANCE_TESTS:BOOL=ON -DVALGRIND_ANALYZE_PERFORMANCE_TESTS:BOOL=ON
+    cmake_extra_flags: -DBUILD_DOCS:BOOL=ON -DTEX_INTERACTION="nonstopmode" -DBUILD_FORTRAN=ON -DBUILD_PACKAGE:BOOL=ON -DBUILD_TESTING:BOOL=ON -DENABLE_REGRESSION_TESTING:BOOL=ON -DREGRESSION_BASELINE_PATH:PATH=$REGRESSION_BASELINE -DREGRESSION_SCRIPT_PATH:PATH=$REGRESSION_DIR -DREGRESSION_BASELINE_SHA:STRING=$REGRESSION_BASELINE_SHA -DCOMMIT_SHA=$COMMIT_SHA -DENABLE_GTEST_DEBUG_MODE:BOOL=OFF -DBUILD_PERFORMANCE_TESTS:BOOL=ON -DVALGRIND_ANALYZE_PERFORMANCE_TESTS:BOOL=ON
```
Comment (Myoldmopar) [inline .decent_ci-Linux.yaml:8]: CI will still default to nonstopmode using this flag

File: CMakeLists.txt [hunk 2]
```diff
@@ -266,4 +266,6 @@ if (BUILD_DOCS)
     # and if it doesn't, clearly state why it fails
     message( FATAL_ERROR "Expected to find xelatex at: ${XELATEX} ; check your LaTeX installation")
   endif()
+  set(TEX_INTERACTION "batchmode" CACHE STRING "Choose the interaction mode for TeX." FORCE)
+  set_property(CACHE TEX_INTERACTION PROPERTY STRINGS "nonstopmode" "batchmode")
```
Comment (Myoldmopar) [inline CMakeLists.txt:270]: Developers will now default to batchmode for a cleaner stdout during builds, but they can switch to nonstopmode easily.

File: cmake/BuildDocumentation.cmake [hunk 3]
```diff
@@ -7,7 +7,7 @@
 # this will get more complicated as we add things like bibtex and makeindex, for now just copy the files into the build tree, execute xelatex twice and rename the output
 #message("Hoping to copy from ${ORIGINAL_CMAKE_SOURCE_DIR}/doc/${INNAME} to ${ORIGINAL_CMAKE_BINARY_DIR}/doc-build/${INNAME}")
 execute_process( COMMAND cmake -E copy_directory ${ORIGINAL_CMAKE_SOURCE_DIR}/doc/${INNAME} ${ORIGINAL_CMAKE_BINARY_DIR}/doc-build/${INNAME})
-execute_process( COMMAND ${XELATEX} -interaction=nonstopmode ${INNAME}.tex )
-execute_process( COMMAND ${XELATEX} -interaction=nonstopmode ${INNAME}.tex )
-execute_process( COMMAND ${XELATEX} -interaction=nonstopmode ${INNAME}.tex )
+execute_process( COMMAND ${XELATEX} -interaction=${TEX_INTERACTION} ${INNAME}.tex )
```
Comment (Myoldmopar) [inline cmake/BuildDocumentation.cmake:10]: This is now no longer hardcoded, but instead relies on the caller to set the value

File: doc/acknowledgments/CMakeLists.txt [hunk 4]
```diff
@@ -22,7 +22,7 @@ SET( OUTPUT_FILENAME "Acknowledgments" )
 FILE(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/doc-build/${SOURCE_FILENAME})
 
 add_custom_command( OUTPUT ${CMAKE_BINARY_DIR}/doc-build/${OUTPUT_FILENAME}.pdf
-	COMMAND ${CMAKE_COMMAND} -DXELATEX=${XELATEX} -DINNAME=${SOURCE_FILENAME} -DOUTNAME=${OUTPUT_FILENAME} -DORIGINAL_CMAKE_SOURCE_DIR=${CMAKE_SOURCE_DIR} -DORIGINAL_CMAKE_BINARY_DIR=${CMAKE_BINARY_DIR} -P ${CMAKE_SOURCE_DIR}/cmake/BuildDocumentation.cmake
+	COMMAND ${CMAKE_COMMAND} -DXELATEX=${XELATEX} -DINNAME=${SOURCE_FILENAME} -DOUTNAME=${OUTPUT_FILENAME} -DORIGINAL_CMAKE_SOURCE_DIR=${CMAKE_SOURCE_DIR} -DORIGINAL_CMAKE_BINARY_DIR=${CMAKE_BINARY_DIR} -DTEX_INTERACTION=${TEX_INTERACTION} -P ${CMAKE_SOURCE_DIR}/cmake/BuildDocumentation.cmake
```
Comment (Myoldmopar) [inline doc/acknowledgments/CMakeLists.txt:25]: Each of these are simply changes to pass the new interaction mode into the cmake script.

**Review by Myoldmopar — APPROVED** [review]
I think this is good to go now.  I am building it on my machine with no issues so far, and a much cleaner stdout.

**Comment by michael-okeefe** [comment]
That's, great, Edwin. Perfect! It's a little thing but I'm excited for this. Thank you.

**Comment by Myoldmopar** [comment]
Sure, as long as the Linux CI builds them fine then I'll merge.

**Comment by Myoldmopar** [comment]
Great, CI built the docs just fine, no issues.  Merging.

**Pull request merged**